packages/jira-adapter/src/filters/workflow_scheme.ts

```diff
@@ -140,6 +140,39 @@ export const preDeployWorkflowScheme = async (
   }
 }
 
+const replaceIDsWithNames = (messages: string[], ids: string[],
+  idToInstance: Record<string, InstanceElement>):string[] => {
+  const newMessages = messages
+  messages.forEach((_message, i) => {
+    ids.forEach(id => {
+      newMessages[i] = newMessages[i].replace(id, idToInstance[id].elemID.getFullName())
+    })
+    newMessages[i] = newMessages[i].replace(new RegExp('ID', 'g'), 'name')
+  })
+  return newMessages
```

Comment:
(besides the comment above)
1. coding-wise:
we try to avoid in-place changes and prefer the functional style (and `string.replace` does not replace in-place anyway), wdyt about:
```suggestion
  return messages.map(m => m
    .replace(id, idToInstance[id].elemID.getFullName())
    .replace(new RegExp('ID', 'g'), 'name')
  )
```
2. result-wise - the elem id is not necessarily the name in the service, so this is a little misleading. can you please align with @liorn on the output format (and document it in the ticket as well)?
3. what about edge cases (e.g. the id was not found in our mapping)?

Reply:
I missed the id loop 🤦‍♀️  so either something of the form
```suggestion
  return messages.map(m => {
    let newMsg = m
    ids.forEach(...) // like you do today
    return newMsg
)
```
or (functional but slightly less readable, so not sure it's better)
```suggestion
  return messages
    .map(msg => _.reduce(ids, (m, id) => m.replace(id, idToInstance[id].elemID.getFullName()), msg))
    .map(msg => msg.replace(new RegExp('ID', 'g'), 'name'))

```

Reply:
I chose the second way, do you think different?